tap_hubspot/automation_streams.py

```diff
@@ -0,0 +1,66 @@
+"""Stream type classes for tap-hubspot."""
+# from black import Report
+from math import inf
+import requests
+import singer
+import json
+
+from dateutil import parser
+import datetime, pytz
+import time
+from datetime import datetime
+
+from pathlib import Path
+from typing import Any, Dict, Optional, Union, List, Iterable
+
+from singer_sdk import typing as th  # JSON Schema typing helpers
+from pathlib import Path
+from typing import Any, Dict, Optional, Union, List, Iterable
+
+from memoization import cached
+
+from singer_sdk.helpers.jsonpath import extract_jsonpath
+from singer_sdk.streams import RESTStream
+from singer_sdk.authenticators import BearerTokenAuthenticator
+from singer_sdk import typing as th  # JSON schema typing helpers
+from tap_hubspot.client import HubspotStream
+from tap_hubspot.streams import ContactsStream
+
+SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
+
+LOGGER = singer.get_logger()
+utc=pytz.UTC
+
+from tap_hubspot.schemas.automation import (
+    Workflows,
+)
+class AutomationStream(HubspotStream):
+    records_jsonpath = "$.workflows[*]"  # Or override `parse_response`.
+    schema_filepath = ""
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of result rows."""
+        self.total_emails = response.json()['workflows']
+        data = response.json()
+        ret = [dict(d, updatedAt=datetime.fromtimestamp(d["updatedAt"]/1000, tz=utc)) for d in data["workflows"]]
```

Comment:
If we are always doing the same `updatedAt` transformation for a lot of Hubspot tables, it might be worth to extract this into a function in a `utils.py` file, or on your base `HubspotStream` class

Reply:
I agree, but I'm having a crisis of philosophy on how to deal with these values.

Technically, a lot of them come in as epochs, or poorly written timestamps (YYYY-MM-DD), so our default state management clause fails. I'm just unsure of how I want to deal with different formats of incoming date-time items.

Option 1) Do nothing, and always change our post_process close to adapt to the varying data -> This may mean we'll have to typecast the data later in the warehouse for ease of understanding for BI (epochs are quite weird)
Option 2) Typecast in parse_response so that every datetime item is sent to the warehouse properly

In terms of best practices, I prefer option 1, but it does duplicate work (need to typecast in post process, and then again change data in warehouse between staging and dbt_prod)